### ISSUE TITLE
[REQUEST] Replace Magic Missile Code

### ISSUE THREAD

**Issue by kepesk1** [Rotwood-Vale/Ratwood-Keep#123]
Magic Missile code causes fireballs to be entity-seeking rather than directly targeting where you click. This sucks and causes unexpected killing of friendlies en masse because most people have no idea this will happen if they've never done it before, and also still happens even if you're fully aware you have basically a 50/50 chance it'll hit where you aim.

Makes mage virtually useless sometimes and a very high liability. Causes vamp lords to kill their whole team by accident. Causes court mage to kill their whole team by accident.

Just make it so magic spells have the same options as bows: arch, and fire. **Arch** to throw a fireball over enemies to a target ON THE GROUND (and not have it fly into the distance until striking something like it does now), and **fire** to shoot a fireball in a straight line until it contacts a target (preferably with a falloff shortly beyond view distance).

TL;DR spells should go where you aim and hit things in-between in a straight line, not auto aim path to the person closest to your cursor.

**Fireballs also do not show any effects when changing Z level to my knowledge, but still explode.

if you recreate the below in-game it will always target the closer entity to both your cursor and your character (the knight). this is a recreation of a real event and i hate it.
![image](https://github.com/Rotwood-Vale/Rotwood-Vale/assets/29184470/2a9ac106-9a02-42d9-8333-ec4fd88747f6)
